### ISSUE TITLE
[V1] [Proposal] clearable & clear-value with model

### ISSUE THREAD

**Issue by olegvakulenko** [quasarframework/quasar#4182]
@rstoenescu 

Before v1 there is exist property `clear-value` and it behavior discussed in issue:
https://github.com/quasarframework/quasar/issues/1994

In v1 when using `clearable` property model set to `null` value only.
For emulate desirable behavior now, if I want to set other clear value, I must add my own action in slot.

When I ask you answered, that `clear-value` property removed for performance reason and I agree with you that permanent checking has some cost. But when I want that kind of opportunity I agreed with performance penalty.

I propose another version of handling `clear-value` property when needed only model for clear value, so if this property defined comparison will be made with two models. When `clear-value` not defined used `null` value.

In other words I propose to make out-of-box solution for this pattern:

```
<q-input v-model="text">
  <template v-if="text !== text$clear" v-slot:append>
    <q-icon name="cancel" @click.stop="text = text$clear" class="cursor-pointer"></q-icon>
  </template>
</q-input>
```

Will be replaced with:

`<q-input clearable clear-value="text$clear" v-model="text">`

or 

`<q-input clearable :clear-value="text$clear" v-model="text">`

(not sure with suitable syntax)
Or use property with name `clear-model` (`:clear-model`).

Same for `QSelect`, where I have custom value for default option.

https://codepen.io/anon/pen/byYmje?editors=1010

What do you think about implementation of this pattern?

**Comment by pdanpdan** [comment]
You can use `@clear` event and set the model to the desired value.
`@clear` event is emitted after the `@input` one that sets the model to null, so the final value of the model will be the one you set on `@clear`.

**Comment by olegvakulenko** [comment]
@pdanpdan 
I understand that with `@clear` event I can set my own default value but in `QSelect` if my clear value not null or not blank, clear icon still visible. In other word I can't get out of box solution when clear icon were hide in case of assign my own default value.
Does my suggestion complicate the logic of work too much or is there a conflict in the settings?

**Comment by pdanpdan** [comment]
Ah, I understand the problem now, sorry - reopening and waiting for @rstoenescu opinion on this.

**Comment by olegvakulenko** [comment]
@pdanpdan 

Hi.
As I can see, PR5756 merged (https://github.com/quasarframework/quasar/pull/5756).
I try to test changes, but see strange behavior. When use clear button on field with slot all work fine, but when use clear button on field with `clearable` property model does't updated correctly, so result of `text !== text$clear` not changed.

And as I can understand PR work only for QInput, not QSelect. Could you please check.

https://codepen.io/olegvakulenko/pen/bGNVYmO?editors=1010

**Comment by olegvakulenko** [comment]
@pdanpdan 

Is it possible to return to the solution of this issue?
I checked v1.9.0, but the problem remained.

**Comment by rstoenescu** [comment]
I'm sorry but we are trying to keep performance at its best. Adding back this pre v1.0 feature would open a can of worms. As long as it can be done through slots, this is the most efficient way. If there's too much "boilerplate" then a component can be created in devland that wraps the desired functionality.

**Comment by olegvakulenko** [comment]
I understand and agreed the desire to achieve maximum framework performance. I just want to clarify whether there is an internal error in the fact that after clicking on the field with cleaning button the condition (`text !== text$clear`) does not work. In other words, the value of the reactive variable is set correctly, but for some reason this does not affect the entire system (the result of condition in the slot and in curly brackets does not change).
Are these features of Quasar or is there still a mistake somewhere?

**Comment by rstoenescu** [comment]
This sounds like it's a mistake in the template or component methods. Highly doubt it's a Vue or Quasar issue. Can you put a codepen in place with the exact scenario above where you get stuck pls?

**Comment by pdanpdan** [comment]
I think there is a misunderstanding: the `clear-value` prop was removed, and the solution is to use the custom slot to put the clear button. So in the codepen all examples using `clear-value` would in fact use `null` as a clear value.

**Comment by olegvakulenko** [comment]
https://codepen.io/olegvakulenko/pen/bGNVYmO?editors=1010

1. Fill filed `1a. Input (slot)`.

```
text: 'qqq';
text !== text$clear: true
```

2. Click clear button in this field.

```
text: '';
text !== text$clear: false
```

Clear button disappear in BOTH input fields.

3. Fill filed `1b. Input (clearable & clear-value)`

```
text: 'qqq';
text !== text$clear: true
```

4. Click clear button in this field.

```
text: '';
text !== text$clear: true
```

We can see that variable `text` got new value, but result of condition `text !== text$clear` NOT change and clear button disappear ONLY in in field `1b`.

**Comment by olegvakulenko** [comment]
Hmm... I somehow missed deleting the parameter `clear-value`.

**Comment by olegvakulenko** [comment]
I can’t insist on the need for such functionality, but for me personally it would greatly simplify writing code for filtering forms, where you need a simple way to reset the value to the default and especially for the QSelect component.
In any case, thanks for the explanation.
